### PR TITLE
Fix sort order in LoadEventNexus

### LINKED ISSUE #32495: LoadEventNexus is labelling event lists as sorted when they're not
If you run LoadEventNexus without compression, it labels all the event lists as sorted on pulse time even if they're not

I noticed this when doing some work on slicing up an event file for the POLARIS instrument. They want to use the FilterByTime algorithm to slice up some event mode data files that were captured over a 15 minute period into smaller ~30 second chunks. The sample used to capture these data files was undergoing a reaction during the 15 minute period so they want to analyse the difference in behaviour over time.

LoadEventNexus sets the sort order in two stages. It first sets up all the event lists for all the spectra with sort order PULSETIME_SORT initially before any data is loaded. This comment explains why:
```
  // Set all (empty) event lists as sorted by pulse time. That way, calling
  // SortEvents will not try to sort these empty lists.
  for (size_t i = 0; i < m_ws->getNumberHistograms(); i++)
    m_ws->getSpectrum(i).setSortOrder(DataObjects::PULSETIME_SORT);
```

The idea seems to be that later on the spectra that actually get some events loaded have their sort order overwritten by the following code in ProcessBankData.cpp:
```
  //------------ Compress Events (or set sort order) ------------------
  // Do it on all the detector IDs we touched
  if (compress) {
    for (detid_t pixID = m_min_id; pixID <= m_max_id; ++pixID) {
      if (usedDetIds[pixID - m_min_id]) {
        // Find the the workspace index corresponding to that pixel ID
        size_t wi = getWorkspaceIndexFromPixelID(pixID);
        auto &el = outputWS.getSpectrum(wi);
        if (compress)
          el.compressEvents(alg->compressTolerance, &el);
        else {
          if (pulsetimesincreasing)
            el.setSortOrder(DataObjects::PULSETIME_SORT);
          else
            el.setSortOrder(DataObjects::UNSORTED);
        }
      }
    }
  }
```
I think the problem is that the surrounding if (compress) statement shouldn't be there. I guess if the events are actually sorted on pulse time in the nexus file then this bug doesn't matter. It's only when the code should be going into the branch where the sort order is set to UNSORTED that the problem exists

### PR DESCRIPTION
**Pull request by DannyHindson**
**Description of work.**

When LoadEventNexus algorithm is run without compression and the events in the Nexus file are not already sorted by pulse time, the algorithm is incorrectly labelling them as sorted by pulse time. This change fixes the problem by always running the code that sets the sort order even if loading without compression (more detail on linked issue)

This problem was causing FilterByTime to incorrectly slice up an event file by time. A lot of events were being assigned to the last second of a run.

I did a few measurements before and after on the load time for a POLARIS nexus file (run 128780) containing 480 million events. There wasn't an increase in the load time. The code changes just means the sort order is set on each event list (one per spectrum) so perhaps not very expensive. The POLARIS file has ~3000 spectra and a single bank which I think gets loaded in 2 threads (splitProcessing=true)

**To test:**

Run this script. The initial nexus file has ~26.5 million events in it spanning a run lasting about 5 minutes. When FilterByTime is run on the workspace with a timestamp range matching the last 1 second of the run it should return ~50,000 events. Before the fix it was returning ~11 million events:

```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

event_ws = Load(Filename=r'POLARIS00129560.nxs')

print("The total number of events: %i" % mtd["event_ws"].getNumberEvents())

event_ws_last_sec = FilterByTime(event_ws,308,309) #11,501,579 events before vs 51098 after code change
```

Fixes #32495. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DannyHindson** [comment]
@martyngigg - I've created this PR to fix the bug I found in LoadEventNexus while supporting Ron doing his event file slicing. Since I don't do much with event files it feels like I should run this past someone. Do you know who would be generally knowledgeable about LoadEventNexus and might give this a look over to check what I've done seems sensible? Also I've currently assigned it to the release but perhaps we can discuss then once someone has validated the fix is OK

**Review by martyngigg — COMMENTED** [review]
@peterfpeterson I'm not sure who might be best to look at this on your side but Danny found a bug in `LoadEventNexus` when compresssion is turned off and not setting the sort order correctly.
Some files will have caused a crash before Danny's change anyway if you don't use compression and this fixes that too.

I think the changes are fine but is there any cause for concern with anything from your side?

File: Framework/DataHandling/test/LoadEventNexusTest.h [hunk 1]
```diff
@@ -181,6 +181,19 @@ class LoadEventNexusTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS("2010-Mar-25 16:11:51.558003540", filteredLogEndTime.toSimpleString());
   }
 
+  void validate_pulse_time_sorting(EventWorkspace_sptr eventWS) {
```
Comment (martyngigg) [inline Framework/DataHandling/test/LoadEventNexusTest.h:184]: Nice. Thanks for adding this test.

**Review by martyngigg — APPROVED** [review]
I'm happy with these changes.

**Pull request merged**